pkg/handlers/pvchandler.go

```diff
@@ -203,15 +209,34 @@ func (pvcHandler *PvcHandler) createPVStorage(pvc v1.PersistentVolumeClaim, pvDi
 	// Set fstab file
 	file, err := os.OpenFile(fstabPath, os.O_APPEND|os.O_WRONLY|os.O_SYNC, 0755)
 	if err != nil {
-		log.Println("ERROR: Cannot open fstab file: " + fstabPath + " because: " + err.Error()+ "\nCannot save mountpoint!")
+		log.Println("ERROR: Cannot open fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
 		return
 	}
 	defer file.Close()
 	bindMountCommand := fmt.Sprintf("%[1]s %[1]s none bind 0 0\n", pvDirPath)
-	_,err = file.WriteString(bindMountCommand)
+	_, err = file.WriteString(bindMountCommand)
 	if err != nil {
-		log.Println("ERROR: Cannot modify fstab file: " + fstabPath + " because: " + err.Error()+ "\nCannot save mountpoint!")
+		log.Println("ERROR: Cannot modify fstab file: " + fstabPath + " because: " + err.Error() + "\nCannot save mountpoint!")
 		return
 	}
-	log.Println("DEBUG: createPVStorage executor successfull!")
+	log.Println("DEBUG: createPVStorage executor successful!")
+}
+
+func generateRandomSuffix(suffixlength int) string {
+	charPool := []byte("abcdefghijklmnopqrstuvwxyz1234567890")
+	rand.Seed(time.Now().Unix())
+	bytes := make([]byte, suffixlength)
+	for i := range bytes {
+		bytes[i] = charPool[rand.Intn(len(charPool))]
+	}
+	return string(bytes)
+}
+
+func isChangeEnoughToProceed(oldPvc v1.PersistentVolumeClaim, newPvc v1.PersistentVolumeClaim) bool {
+	if !reflect.DeepEqual(oldPvc, v1.PersistentVolumeClaim{}) {
+		if len(newPvc.ObjectMeta.Finalizers) > 0 && !reflect.DeepEqual(oldPvc.ObjectMeta.Finalizers, newPvc.ObjectMeta.Finalizers) && oldPvc.Spec.VolumeName == newPvc.Spec.VolumeName && newPvc.Spec.VolumeName == "" {
```

Comment:
what's up with the finalizer-obsession? :)

Reply:
ok, I gave up my obsession :)
fixed - without checking the finalizer